controllers/utils/reconciler_utils.go

```diff
@@ -26,6 +28,41 @@ func GetClusterId(ctx context.Context, client k8sclient.Client) (string, error)
 	return string(v.Spec.ClusterID), nil
 }
 
+//returns cluster Openshift version
+func GetClusterOSVersion(ctx context.Context, client k8sclient.Client) (string, error) {
+	v := &v13.ClusterVersion{}
+	selector := k8sclient.ObjectKey{
+		Name: "version",
+	}
+
+	err := client.Get(ctx, selector, v)
+	if err != nil {
+		return "", err
+	}
+	return v.Status.Desired.Version, nil
+}
+
+//returns true if test cluster version string is same or newer than compareTo version string
+func HasNewerOrSameClusterVersion(test string, compareTo string) (bool, error) {
+	var testVersion semver.Version
+	compareToVersion, err := semver.Make(compareTo)
+	if err != nil {
+		return false, err
+	}
+	testVersion, err = semver.Make(test)
+	if err != nil {
+		return false, err
+	}
+
+	if testVersion.Major > compareToVersion.Major {
```

Comment:
I think you could use the semver.Version functions GT here?
e.g. `testVersion.GT(compareToVersion)`

Reply:
@laurafitzgerald I did have problems using semver GTE func here. I used v4.10.0-rc.1 cluster for verification and GTE would return false when compared to 4.10.0. Unsure of better way to compare. Maybe a different compareToVersion case?